server/remote/msgraph/get_me.go

```diff
@@ -37,3 +45,50 @@ func (c *client) GetMe() (*remote.User, error) {
 
 	return user, nil
 }
+
+func (c *client) CheckUserStatus() bool {
+	// Not checking for API calls made using super client
+	if c.store == nil {
+		return true
+	}
+
+	user, err := c.store.LoadUser(c.mattermostUserID)
+	if err != nil {
+		c.Logger.Errorf("Not able to load the user %s. %s", c.mattermostUserID, err.Error())
+		return false
+	}
+
+	// Checking if the user is marked as inactive
+	if user.OAuth2Token.AccessToken == "" {
+		if _, err := c.poster.DM(c.mattermostUserID, ErrorUserInactive); err != nil {
+			c.Logger.Errorf("Not able to DM the user %s. %s", c.mattermostUserID, err.Error())
+		}
+		return false
+	}
+
+	return true
```

Comment:
This and `ChangeUserStatus` seems wrong to be in a file called `server/remote/msgraph/get_me.go`. I think the logic pertaining to the store etc. should be in a different package.